clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/common/model/AzureDeploymentOperation.groovy

```diff
@@ -0,0 +1,150 @@
+import com.netflix.spinnaker.clouddriver.azure.security.AzureCredentials
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import groovy.util.logging.Slf4j
+
+@Slf4j
+class AzureDeploymentOperation {
+  static ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+
+  static AzureDeploymentOperation getObjectFromJson(String responseContent) {
+    mapper.readValue(responseContent, AzureDeploymentOperation.class)
+  }
+
+  /**
+   * Return a collection of deployment related errors (empty list if deployment was successful)
+   * @param task the current task to be updated
+   * @param creds the credentials to use when communicating to the Azure subscription(s)
+   * @param resourceGroupName the resource group where we try to deploy
+   * @param deploymentName the name of the deployment
+   * @return a Collection of error messages capturing deployment related errors
+   */
+  static List<String> checkDeploymentOperationStatus(Task task, String opsName, AzureCredentials creds, String resourceGroupName, String deploymentName) {
+    def errList = new ArrayList<String>()
+    Map<String, Boolean> resourceCompletedState = new HashMap<String, Boolean>()
+    String deploymentState = AzureResourceManagerClient.DeploymentState.DEPLOYING
+    Boolean checkDeployment = true
+
+    while (checkDeployment) {
+      deploymentState = creds.resourceManagerClient.getDeployment(creds, resourceGroupName, deploymentName).properties.provisioningState
+
+      creds.resourceManagerClient.getDeploymentOperations(creds, resourceGroupName, deploymentName).each {DeploymentOperation d ->
+
+        if (!resourceCompletedState.containsKey(d.id)){
+          resourceCompletedState[d.id] = false
+        }
+        if (d.properties.provisioningState == AzureResourceManagerClient.DeploymentState.SUCCEEDED) {
+          if (!resourceCompletedState[d.id]) {
+            task.updateStatus opsName, String.format("Resource %s created", d.properties.targetResource.resourceName)
+            resourceCompletedState[d.id] = true
+          }
+        }
+        else if (d.properties.provisioningState == AzureResourceManagerClient.DeploymentState.FAILED) {
+          if (!resourceCompletedState[d.id]) {
+
+            // Work around for Azure SDK bug while retrieving the status message
+            // TODO remove the REST call and use d.properties.satusMessage.error.message instead
+
+            String responseContent = creds.resourceManagerClient.
+              getAzureRESTJson(
+                              creds,
+                              "https://management.azure.com",
+                              "resourceGroups/${resourceGroupName}/providers/Microsoft.Resources/deployments/${deploymentName}/operations",
+                              ["api-version=2015-11-01"]
+                              )
+            AzureDeploymentOperation updatedDeploymentOperation = getObjectFromJson(responseContent)
```

Comment:
Seems like the kind of method where you should be passing AzureDeploymentOperation into getAzureXXX() and having it do the conversion?